### ISSUE TITLE
App crashes due to permission issue.

### ISSUE THREAD

**Issue by RaghuChandrasekaran** [midhunhk/trip-o-meter#11]
App crashes after clicking on Expenses tab.This is after the new update.
Tried a fresh install too.

Error Log from logcat:
```
 FATAL EXCEPTION: main
                                                   Process: com.ae.apps.tripmeter, PID: 11111
                                                   java.lang.RuntimeException: Unable to start activity ComponentInfo{com.ae.apps.tripmeter/com.ae.apps.tripmeter.activities.MainActivity}: java.lang.SecurityException: Permission Denial: opening provider com.android.providers.contacts.ContactsProvider2 from ProcessRecord{8d29387 11111:com.ae.apps.tripmeter/u0a182} (pid=11111, uid=10182) requires android.permission.READ_CONTACTS or android.permission.WRITE_CONTACTS
                                                       at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2423)
                                                       at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2483)
                                                       at android.app.ActivityThread.access$900(ActivityThread.java:153)
                                                       at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1349)
                                                       at android.os.Handler.dispatchMessage(Handler.java:102)
                                                       at android.os.Looper.loop(Looper.java:148)
                                                       at android.app.ActivityThread.main(ActivityThread.java:5441)
                                                       at java.lang.reflect.Method.invoke(Native Method)
                                                       at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:738)
                                                       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:628)
                                                    Caused by: java.lang.SecurityException: Permission Denial: opening provider com.android.providers.contacts.ContactsProvider2 from ProcessRecord{8d29387 11111:com.ae.apps.tripmeter/u0a182} (pid=11111, uid=10182) requires android.permission.READ_CONTACTS or android.permission.WRITE_CONTACTS
                                                       at android.os.Parcel.readException(Parcel.java:1620)
                                                       at android.os.Parcel.readException(Parcel.java:1573)
                                                       at android.app.ActivityManagerProxy.getContentProvider(ActivityManagerNative.java:3553)
                                                       at android.app.ActivityThread.acquireProvider(ActivityThread.java:4797)
                                                       at android.app.ContextImpl$ApplicationContentResolver.acquireUnstableProvider(ContextImpl.java:2031)
                                                       at android.content.ContentResolver.acquireUnstableProvider(ContentResolver.java:1484)
                                                       at android.content.ContentResolver.query(ContentResolver.java:477)
                                                       at android.content.ContentResolver.query(ContentResolver.java:435)
                                                       at com.ae.apps.common.c.a.a(Unknown Source)
                                                       at com.ae.apps.common.c.a.<init>(Unknown Source)
                                                       at com.ae.apps.tripmeter.d.a.<init>(Unknown Source)
                                                       at com.ae.apps.tripmeter.d.b.<init>(Unknown Source)
                                                       at com.ae.apps.tripmeter.d.b.a(Unknown Source)
                                                       at com.ae.apps.tripmeter.b.h.a(Unknown Source)
                                                       at android.support.v4.b.l.j(Unknown Source)
                                                       at android.support.v4.b.r.a(Unknown Source)
                                                       at android.support.v4.b.r.z(Unknown Source)
                                                       at android.support.v4.b.r.a(Unknown Source)
                                                       at android.support.v4.b.r.b(Unknown Source)
                                                       at android.support.v4.b.r.f(Unknown Source)
                                                       at android.support.v4.b.o.n(Unknown Source)
                                                       at android.support.v4.b.m.onStart(Unknown Source)
                                                       at android.support.v7.app.c.onStart(Unknown Source)
                                                       at android.app.Instrumentation.callActivityOnStart(Instrumentation.java:1238)
                                                       at android.app.Activity.performStart(Activity.java:6320)
                                                       at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2386)
                                                       at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2483) 
                                                       at android.app.ActivityThread.access$900(ActivityThread.java:153) 
                                                       at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1349) 
                                                       at android.os.Handler.dispatchMessage(Handler.java:102) 
                                                       at android.os.Looper.loop(Looper.java:148) 
                                                       at android.app.ActivityThread.main(ActivityThread.java:5441) 
                                                       at java.lang.reflect.Method.invoke(Native Method) 
                                                       at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:738) 
                                                       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:628)
```

**Comment by RaghuChandrasekaran** [comment]
After going through the error log it is understood that the issue is with permission.I manually went into the info page of app and gave permission to access contacts after which this crash issue is resolved.
Shouldn't the app after update or fresh install ask for the permission by itself?

**Comment by midhunhk** [comment]
Do we need to explicitly invoke the Permissions window?

**Comment by RaghuChandrasekaran** [comment]
@midhunhk Yes from API lvl 23 & above the permissions are divided into to Normal and Dangerous.Normal permissions are given automatically (e.g Flashlight) but for Dangerous permissions we need to ask for user to give the required permission otherwise would cause issues like this one.

[Guide](https://developer.android.com/training/permissions/requesting.html)
[Reference](https://developer.android.com/guide/topics/permissions/requesting.html#normal-dangerous)

**Comment by midhunhk** [comment]
Will be fixed with #12

**Comment by midhunhk** [comment]
Fixed by pull request #12
